src/core/worker.c

```diff
@@ -546,145 +548,197 @@ QuicWorkerProcessConnection(
     }
 }
 
-CXPLAT_THREAD_CALLBACK(QuicWorkerThread, Context)
-{
-    QUIC_WORKER* Worker = (QUIC_WORKER*)Context;
+typedef enum QUIC_WORKER_LOOP_STATE {
+    QUIC_WORKER_LOOP_EXIT,
+    QUIC_WORKER_LOOP_CONTINUE,
+    QUIC_WORKER_LOOP_WAIT,
+    QUIC_WORKER_LOOP_WAIT_TIMER,
+} QUIC_WORKER_LOOP_STATE;
 
-    Worker->ThreadID = CxPlatCurThreadID();
-    Worker->IsActive = TRUE;
-    QuicTraceEvent(
-        WorkerStart,
-        "[wrkr][%p] Start",
-        Worker);
+typedef enum QUIC_WORKER_LOOP_REASON {
+    QUIC_WORKER_LOOP_REASON_NORMAL,
+    QUIC_WORKER_LOOP_REASON_WAIT,
+    QUIC_WORKER_LOOP_REASON_TIMER,
+} QUIC_WORKER_LOOP_REASON;
 
+//
+// Runs one iteration of the worker loop.
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_WORKER_LOOP_STATE
+QuicWorkerLoop(
```

Comment:
While not a problem with the current calling pattern, eventually, if we allow an app to choose when to call this function, it may use a single thread to drive multiple workers, and in that case, would probably round robin the workers even when one of them returns `QUIC_WORKER_LOOP_CONTINUE`. I am just wondering now this might effect the `IsActive` and queue delay variables of the worker... I think it's still reasonable to keep things as they are now, but it's worth some additional thought.